src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs

```diff
@@ -118,7 +118,10 @@ public void Format(LogEvent logEvent, TextWriter output)
                 }
 
                 if (pt.Alignment.HasValue)
+                {
                     Padding.Apply(output, ((StringWriter)writer).ToString(), pt.Alignment);
+                    writer.Dispose();
```

Comment:
Why did you place `Dispose` here?

Reply:
So that writer would get back into a pool and be reused.

Reply:
I would move `Dispose` call from inside here and place after all conditions:
```cs
if (writer is ReusableStringWriter)
    writer.Dispose();
```

Reply:
OK

Reply:
Well... I thought about it again. Look - it's very important to call `Dispose`. Otherwise you will lose `ReusableStringWriter._pooledWriter` **forever**. So what I would really like to see here - a guaranteed `Dispose` call.

Reply:
🤦‍♂️ I was wrong. In case of exception you will lose `ReusableStringWriter._pooledWriter` until the next call to `ReusableStringWriter.GetOrCreate`.